### DISCUSSION TITLE
Provide an alert when refreshing YAML if there's a change that requires you to restart Home Assistant instead

### DISCUSSION THREAD

**Discussion by twerner** [home-assistant/frontend#19775] [Other feature requests]
I just added my first RESTful sensor. Adding this required a restart of Home Assistant, while any changes afterwards require just pressing the reload YAML button. I would like a notification to pop up when I try to reload YAML that says something along the lines of "There are pending YAML changes that require a Home Assistant restart to load."

**Comment by ttreker** [comment]
I would like this feature also! It would make it a little more nuanced. Not only should it report that a change requires a restart but it should additionally report which "Reload Config" buttons could have been used to just reload the changes that had been made. It is always unclear whether a change made can be immediately reloaded and what reload button to press if it can. Some buttons are clearer than others as to what they will reload. So here is what I propose:

If you press "All YAML Configuration" It not only reloads all the YAML config it normally would but it also:

- Reports if a start is required
- Lists all reload buttons that could have been pressed so as to avoid having to press "All YAML Configuration" or "Restart Home Assistant".

If it did this one could quickly learn what they can quickly reload with specific buttons and what requires a restart. This would save time during development and testing.
